modules/ServerAPI/src/server/CustomerSpecificServerFunctions.coffee

```diff
@@ -144,24 +144,21 @@ exports.isUserAdmin = (user) ->
 exports.findByUsername = (username, fn) ->
 	return exports.getUser username, fn
 
-formatSystemRolesFromSSOGroups = (ssoGroups) =>
-	# Formats the sso roles into ACAS roles
-	# Case insensitive check for "CMPDREG" in the name assigned the role to the lsKind 'CmpdReg'
-	# if not matched, it assigns to role to the lsKind 'ACAS'
-	roles = []
-	if ssoGroups?
-		if typeof ssoGroups == "string"
-			groups = [ssoGroups]
-		for group in ssoGroups
-			roleName = group.toUpperCase()
-			lsKind = 'ACAS'
-			if roleName.indexOf("CMPDREG") > -1
-				lsKind = 'CmpdReg'
-			roles.push
-				lsType: 'System'
-				lsKind: lsKind
-				roleName: "ROLE_#{roleName}"
+formatSystemRolesFromSSOProfile = (profile) =>
+	# Map Profile Attributes to ACAS/CmpdReg System Roles
+	config = require "#{ACAS_HOME}/conf/compiled/conf.js"
 	
+	# Parse the config which maps the profile attributes to system roles
+	profileGroupToACASRoles = JSON.parse(config.all.server.security.saml.profileAttributeToSystemRoles)
+	lsType = 'System'
```

Comment:
I missed the fact that this was "System" before. Originally I had thought the lsType was supposed to represent the origin of that role, so
"System": ACAS internal role management
"LDAP": from LDAP
"Project": for project roles (no longer used)
Could we change the lsType to "IdP" if that won't break anything? That will then keep it straight which roles came from where.

Reply:
I actually didn't know this and there are several places in the code base where we've set "System" as the lsType.  I actually didn't realize the difference between LDAP and system until now and a little unsure how to proceed.  It's pretty easy to switch everything over to IdP from System, but when I do that, it does actually wipe out all other roles. i.e. I my user had System roles, then I switched everything to IdP and now my user only has IdP roles?

Reply:
I think that is expected. My understanding is that IdP or LDAP roles should take precedence over System roles. So some "internal" users or service accounts could have System roles, but if someone is getting any roles from IdP, we expect that user to get all their roles from the IdP.

Reply:
Resolving this conversation considering changing the lsType as out of scope.